src/modules/lua_mtev.c

```diff
@@ -1064,6 +1070,13 @@ mtev_lua_socket_read_complete(eventer_t e, int mask, void *vcl,
   int len;
   int args = 0;
 
+  if(cl->timeout_event) {
+    eventer_remove_timed(cl->timeout_event);
+    lua_timeout_callback_ref_free(cl->timeout_event->closure);
```

Comment:
deregister the event with lua here. (you need the mtev_lua_resume_info_t to do that)